File: frontend/src/app/main/dto/game.ts
import { Field } from "src/app/board/dto/field";

export interface Game {
    id: number;
    invitation: "Issued" | "Accepted" | "Rejected";
    gameType: "User" | "AI";
    aiType: "Random" | "None";
    gameStatus: "NotFinished" | "Won" | "Lost" | "Drawn";
    currentState: Field[][];
    lastMoveRow: number; // TODO
    lastMoveColumn: number; // TODO
    userStarts: boolean;
    userTurn: boolean;
    user: String;
    opponent: String; 
    user_id: number;
    opponent_id: number; 
    invert: boolean;
}

File: main/src/main/java/io/github/xpakx/chess/game/Game.java
package io.github.xpakx.chess.game;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.github.xpakx.chess.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Game {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private InvitationStatus invitation;
    private GameType type;
    private AIType aiType;
    private GameStatus status;

    @Column(columnDefinition = "TIMESTAMP")
    private LocalDateTime startedAt;
    @Column(columnDefinition = "TIMESTAMP")
    private LocalDateTime lastMoveAt;

    private String currentState;
    private Integer nonCaptureMoves;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "opponent_id")
    @JsonIgnore
    private User opponent;

    private boolean userStarts;
    private boolean userTurn;
}
